client.go

```diff
@@ -80,6 +80,10 @@ type Client struct {
 	protocol    Protocol
 	logger      hclog.Logger
 	doneCtx     context.Context
+
+	// NegotiatedVersion is a record of the protocol version that was returned
+	// by the server.
+	NegotiatedVersion int
```

Comment:
Just in the style of this particular library, I'd prefer this be unexported and an accessor be given: `client.NegotiatedVersion()`. I would also document that this is only available after Start is called.